runtime/src/bank.rs

```diff
@@ -779,9 +779,7 @@ impl Bank {
             slots_per_year: parent.slots_per_year,
             epoch_schedule,
             collected_rent: AtomicU64::new(0),
-            rent_collector: parent
-                .rent_collector
-                .clone_with_epoch(epoch, parent.cluster_type()),
+            rent_collector: parent.rent_collector.clone_with_epoch(epoch),
```

Comment:
change of plumbing by removing `RentCollecotr::cluster_type` wasn't avoidable exactly because of unavailability of feature set of new bank even before creating it.